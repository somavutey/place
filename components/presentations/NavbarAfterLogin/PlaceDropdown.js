import React, { useState } from "react";
import { placeDropdown } from "./DropdownItem"; //y
import Link from "next/link";
import { makeStyles } from "@mui/styles";
import { fireAuth } from '../../../services/firebase'

const useStyle = makeStyles({
  servicesSubmenu: {
    position: "absolute",
    top: "8vh",
    listStyle: "none",
    textAlign: "center",
    padding: 0,
    zIndex: '1',
  },

  li: {
    display: "block",
    width: "100px",
    height: "100%",
    textDecoration: "none",
    color: "white",
    background: "rgb(0, 212, 212)",
    cursor: "pointer",
    border: "1px solid",
    padding: "10px",
    "&:hover": {
      background: "rgb(0, 175, 175)",

    },
  },

  servicesSubmenuClicked: {
    display: "none",
  },
});
// use function in utils folder
export default function PlaceDropdown() {
  const classes = useStyle();
  const [place, setPlace] = useState(false);
  const handleLogout = (e) => {
    e.preventDefault();
    fireAuth.signOut()
      .then((res) => {
        alert("Signed Out")
      })
      .catch((err) => {
        alert(err.message)
      }
      )
  }
  return (
    <>
      <ul
        className={
          place ? classes.servicesSubmenuClicked : classes.servicesSubmenu
        }
        onClick={() => setPlace(!place)}
      >
        {placeDropdown.map((item) => {
          return (
            <li className={classes.li} key={item.id}>
              <Link
                href={item.path}
                className={classes.submenuItem}
                onClick={() => setPlace(false)}
              >
                {item.title}
              </Link>
            </li>
          );
        })}
        <li className={classes.li} key="oo1">
          <Link >Logout</Link>
        </li>
      </ul>
    </>
  );
}
