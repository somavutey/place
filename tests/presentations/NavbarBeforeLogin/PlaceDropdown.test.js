import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import PlaceDropdown from "./../../../components/presentations/NavbarBeforeLogin/PlaceDropdown";

it("PlaceDropdown", () => {
  const wrapper = shallow(<PlaceDropdown />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
