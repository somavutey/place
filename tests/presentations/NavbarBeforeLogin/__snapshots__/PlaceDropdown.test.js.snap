// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceDropdown 1`] = `
<Fragment>
  <ul
    className="makeStyles-servicesSubmenu-1"
    onClick={[Function]}
  >
    <li
      className="makeStyles-li-2"
      key="1"
    >
      <Link
        href="./../../../FilterPagesAfterLogin"
        onClick={[Function]}
      >
        Outdoor
      </Link>
    </li>
    <li
      className="makeStyles-li-2"
      key="2"
    >
      <Link
        href="./../../../FilterPagesAfterLogin"
        onClick={[Function]}
      >
        Indoor
      </Link>
    </li>
    <li
      className="makeStyles-li-2"
      key="3"
    >
      <Link
        href="./../../../FilterPagesAfterLogin"
        onClick={[Function]}
      >
        Modern
      </Link>
    </li>
    <li
      className="makeStyles-li-2"
      key="4"
    >
      <Link
        href="./../../../FilterPagesAfterLogin"
        onClick={[Function]}
      >
        Natural
      </Link>
    </li>
  </ul>
</Fragment>
`;
